sdk/trace/trace_test.go

```diff
@@ -1972,15 +1972,98 @@ func TestWithIDGenerator(t *testing.T) {
 
 			gotSpanID, err := strconv.ParseUint(span.SpanContext().SpanID().String(), 16, 64)
 			require.NoError(t, err)
-			assert.Equal(t, uint64(startSpanID+i), gotSpanID)
+			assert.Equal(t, uint64(startSpanID)+uint64(i), gotSpanID)
 
 			gotTraceID, err := strconv.ParseUint(span.SpanContext().TraceID().String(), 16, 64)
 			require.NoError(t, err)
-			assert.Equal(t, uint64(startTraceID+i), gotTraceID)
+			assert.Equal(t, uint64(startTraceID)+uint64(i), gotTraceID)
 		}()
 	}
 }
 
+func TestIDsRoundTrip(t *testing.T) {
+	gen := defaultIDGenerator()
+	n := 1000
+
+	for i := 0; i < n; i++ {
+		traceID, spanID := gen.NewIDs(context.Background())
+		gotTraceID, err := trace.TraceIDFromHex(traceID.String())
+		assert.NoError(t, err)
+		assert.Equal(t, traceID, gotTraceID)
+		gotSpanID, err := trace.SpanIDFromHex(spanID.String())
+		assert.NoError(t, err)
+		assert.Equal(t, spanID, gotSpanID)
+	}
+}
+
+func TestIDConversionErrors(t *testing.T) {
+	for _, tt := range []struct {
+		name         string
+		spanIDStr    string
+		traceIDStr   string
+		spanIDError  string
+		traceIDError string
+	}{
+		{
+			name:         "valid ids",
+			spanIDStr:    sid.String(),
+			traceIDStr:   tid.String(),
+			spanIDError:  "",
+			traceIDError: "",
```

Comment:
I think it would be better if this would be a separate unit test so that we do not need to have `if-else` in all assertions just for this one test case.

Reply:
FWIW this is how I typically structure a test, checking both positive and negative cases together.
But since you don't like it, I removed this case entirely; these IDs come from `TraceIDFromHex` and there is a round-trip test already.